modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ParameterGroupAwareObjectBuilder.java

```diff
@@ -58,11 +57,9 @@ public T build(ResolverSetResult result) throws MuleException {
 
   private List<ValueSetter> createSingleValueSetters(Class<?> prototypeClass, ResolverSet resolverSet) {
     return resolverSet.getResolvers().keySet().stream().map(parameterModel -> {
-      Optional<Field> field = getField(prototypeClass, parameterModel);
-
       // if no field, then it means this is a group attribute
-      return field.isPresent() ? new SingleValueSetter(parameterModel, field.get()) : null;
-    }).filter(field -> field != null).collect(new ImmutableListCollector<>());
+      return getField(prototypeClass, parameterModel).map(f -> new SingleValueSetter(parameterModel, f));
+    }).filter(Optional::isPresent).map(Optional::get).collect(new ImmutableListCollector<>());
```

Comment:
the optional of an optional is not a good idea, and you're not handling this in the nicest way. Go back to the previous approach, at least it was more clear